### PR TITLE
Add shared_memory, memory_maps, process_memory_map table to Linux

### PR DESCRIPTION
**Pull request by theopolis**

### CODE REVIEW

File: osquery/tables/system/linux/shared_memory.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    // Accessor, creator pids.
+    r["pid"] = BIGINT(shmseg.shm_lpid);
+    r["creator_pid"] = BIGINT(shmseg.shm_cpid);
+
+    // Access, detached, creator times
+    r["atime"] = BIGINT(shmseg.shm_atime);
+    r["dtime"] = BIGINT(shmseg.shm_dtime);
+    r["ctime"] = BIGINT(shmseg.shm_ctime);
+
+    r["permissions"] = INTEGER(ipcp->mode);
```
Comment (theopolis) [inline osquery/tables/system/linux/shared_memory.cpp]: This should use the `lsperms` method defined in file. As an integer this is hard to extract meaning from.

**Comment by theopolis** [comment]
Implements the removal of `path` and `name` from `process_envs` from #721

File: osquery/tables/system/linux/shared_memory.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  unsigned long swap_attempts;
+  unsigned long swap_successes;
+};
+
+QueryData genSharedMemory(QueryContext &context) {
+  QueryData results;
+
+  // Use shared memory control (shmctl) to get the max SHMID.
+  struct shm_info shm_info;
+  int maxid = shmctl(0, SHM_INFO, (struct shmid_ds *)(void *)&shm_info);
```
Comment (theopolis) [inline osquery/tables/system/linux/shared_memory.cpp]: Tthis is not doing anything with `shm_info`.

**Comment by javuto** [comment]
@osqueryer, ok to test

**Pull request merged**